core/src/main/scala/org/apache/spark/deploy/history/EventLogFileWriters.scala

```diff
@@ -229,6 +232,15 @@ class SingleEventLogFileWriter(
     writeLine(eventJson, flushLogger)
   }
 
+  override def writeToXAttr(attrName: String, attrValue: String): Unit = {
+    try {
+      fileSystem.setXAttr(new Path(inProgressPath), attrName, attrValue.getBytes())
+    } catch {
+      case _: IOException =>
```

Comment:
hmm, I think additional catch `UnsupportedOperationException` should be added. Not all `FileSystem` supports this api, the default  implementation is to throw `UnsupportedOperationException`.

https://github.com/apache/hadoop/blob/e22f5e75aeff23d6c8dc1b67cb6f4fd4a4c416dc/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/  .java#L3225-L3229